skin/src/main/java/org/researchstack/skin/schedule/ScheduleHelper.java

```diff
@@ -14,15 +18,37 @@
 {
     private ScheduleHelper() {}
 
-    public static Date nextSchedule(String cronString, Date lastExecution)
+    /**
+     *
+     * @param cronString the cron string that describes the periodicity
+     * @param delay delay in ISO 8601 since the scheduling was actually started the first time (after onboarding)
+     * @param interval minimal amount of time between two executions, complements the chron information
+     * @param usestart when the app was started first, should be after onboarding and never null
+     * @param lastExecution last time the task was completed, can be null
+     * @return
+     */
+    public static Date nextSchedule(String cronString, String delay, String interval, @NonNull Date usestart, Date lastExecution)
```

Comment:
My fault, you don't have access to the Jira obviously.

Just wanted to share we are planning on re-writing the ScheduleHelper soon, so any changes to it should be able to go in.